common/read_only_map.go

```diff
@@ -24,3 +26,7 @@ func (m *ReadOnlyMap[K, V]) Keys() []K {
 func (m *ReadOnlyMap[K, V]) Len() int {
 	return len(m.data)
 }
+
+func (m *ReadOnlyMap[K, V]) Equal(data map[K]V) bool {
+	return reflect.DeepEqual(m.data, data)
```

Comment:
Reflection may have perf issue, so may be not a good choice for non testing use.

Reply:
Good point